vod/hls/m3u8_builder.c

```diff
@@ -1286,6 +1368,18 @@ m3u8_builder_build_master_playlist(
 		max_video_stream_inf += sizeof(M3U8_STREAM_TAG_SUBTITLES) - 1 + VOD_INT32_LEN;
 	}
 
+	if (media_set->closed_captions == NULL)
```

Comment:
this should be aligned with the write code - 
if (media_set->closed_captions < media_set->closed_captions_end) { ... } 
else if (media_set->closed_captions != NULL) { ... }